diploma-project/pages/catalog/[id].jsx

```diff
@@ -0,0 +1,243 @@
+import {StandardLayout} from "../../components";
+import {default_profile_picture} from "../../public/constants";
+import Image from "next/future/image";
+import {useEffect, useState} from "react";
+import {getCatalog, Product} from "../../services/product";
+import {ProductCard} from "../../components";
+import {useRouter} from "next/router";
+import {getUserById} from "../../services/user";
+
+const Catalog = ({user}) => {
+    const [products, setProducts] = useState(null);
+    const router = useRouter();
+
+    useEffect(() => {
```

Comment:
ditto. Depending on your use case, if you want to refetch the catalog when user.catalog changes, you should call getCatalog with user.catalog as an argument.

Reply:
I have `user.catalog` as a dependency in my `useEffect` so I will refetch the catalog if there are changes. There is a difference between `user.catalog` and `catalog` variables. The `user.catalog` is a list of IDs whereas `catalog` is a list of products. I will rename it so it becomes more clear - `catalog` to `products`.